tests/lit-tests/ptr-assign-inst.ispc

```diff
@@ -0,0 +1,26 @@
+// This test checks that we can properply instantiate a template function at
+// the moment when we assign its address to a function pointer.
+
+// RUN: %{ispc} --target=host --nostdlib %s --emit-llvm-text -o - 2>&1 | FileCheck %s
+
+float same_exp(float x);
+uniform int same_exp(uniform int x);
+
+typedef float (*FuncType)(float);
+typedef uniform int (*FuncType2)(uniform int);
```

Comment:
So looks like you're adding support only for one case and the code below would not compile?
```
typedef uniform double (FuncType3)(uniform double);
uniform double foo3(uniform double a) {
    uniform* FuncType3 func_ptr = same;
    return func_ptr(a);
}

Reply:
The syntax is incorrect. For example, consider `const *int x`. Therefore, it should be `uniform FuncType3 *uniform func_ptr`, and it works as demonstrated by the test case I have added.